### PR TITLE
Run multiple go-ethereum nodes in e2e

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

E2E feature

**What does this PR do? Why is it needed?**

This PR splits the existing eth1 code into three separate types:
- `Miner` responsible for mining blocks and deploying the deposit contract
- `Node` responsible for connecting to the miner and other nodes through the P2P network
- `NodeSet` responsible for managing a collection of `Node` objects

The reason for this split is to allow running multiple Eth1 nodes in E2E. We connect each beacon node to a separate Eth1 node (including the mining node) to represent a CL+EL setup resembling the real system.

**Which issues(s) does this PR fix?**

Part of #10228

### CODE REVIEW

File: config/params/testnet_e2e_config.go [hunk 1]
```diff
@@ -42,8 +42,8 @@ func E2ETestConfig() *BeaconChainConfig {
 	e2eConfig.MaxSeedLookahead = 1
 
 	// PoW parameters.
-	e2eConfig.DepositChainID = 1337   // Chain ID of eth1 dev net.
-	e2eConfig.DepositNetworkID = 1337 // Network ID of eth1 dev net.
+	e2eConfig.DepositChainID = 123456   // Chain ID of eth1 dev net.
```
Comment (nisdas) [inline config/params/testnet_e2e_config.go:45]: lets change this back to 1337 and use the value of `1337` in the genesis config

File: testing/endtoend/components/lighthouse_beacon.go [hunk 2]
```diff
@@ -112,7 +112,7 @@ func (node *LighthouseBeaconNode) Start(ctx context.Context) error {
 		fmt.Sprintf("--port=%d", e2e.TestParams.BeaconNodeRPCPort+index+e2e.LighthouseP2PPortOffset),
 		fmt.Sprintf("--http-port=%d", e2e.TestParams.BeaconNodeRPCPort+index+e2e.LighthouseHTTPPortOffset),
 		fmt.Sprintf("--target-peers=%d", 10),
-		fmt.Sprintf("--eth1-endpoints=http://127.0.0.1:%d", e2e.TestParams.Eth1RPCPort),
+		fmt.Sprintf("--eth1-endpoints=http://127.0.0.1:%d", e2e.TestParams.Eth1RPCPort+10*node.index),
```
Comment (nisdas) [inline testing/endtoend/components/lighthouse_beacon.go:115]: why the change ?

File: testing/endtoend/components/validator.go [hunk 3]
```diff
@@ -229,19 +229,12 @@ func SendAndMineDeposits(keystorePath string, validatorNum, offset int, partial
 	if err = sendDeposits(web3, keystoreBytes, validatorNum, offset, partial); err != nil {
 		return err
 	}
-	mineKey, err := keystore.DecryptKey(keystoreBytes, "" /*password*/)
```
Comment (nisdas) [inline testing/endtoend/components/validator.go:232]: why not add this back ?

File: testing/endtoend/endtoend_test.go [hunk 4]
```diff
@@ -88,36 +89,41 @@ func (r *testRunner) run() {
...
-		if err := eth1Node.Start(ctx); err != nil {
-			return errors.Wrap(err, "failed to start eth1node")
+		if err := bootNode.Start(ctx); err != nil {
+			return errors.Wrap(err, "failed to start bootnode")
 		}
 		return nil
 	})
+
+	// ETH1 nodes.
+	eth1Nodes := eth1.NewNodeSet()
```
Comment (nisdas) [inline testing/endtoend/endtoend_test.go:102]: Don't think the bootnode is required here

File: testing/endtoend/minimal_e2e_test.go [hunk 5]
```diff
@@ -78,7 +78,7 @@ func e2eMinimal(t *testing.T, args *testArgs) {
 		ev.ValidatorsVoteWithTheMajority,
 		ev.ColdStateCheckpoint,
 		ev.ForkTransition,
-		ev.APIMiddlewareVerifyIntegrity,
+		//ev.APIMiddlewareVerifyIntegrity,
```
Comment (rkapka) [inline testing/endtoend/minimal_e2e_test.go:81]: Uncomment

**Review by nisdas — APPROVED** [review]

**Review by nisdas — APPROVED** [review]
